bundles/org.eclipse.swt/Eclipse SWT/gtk/org/eclipse/swt/widgets/DateTime.java

```diff
@@ -1868,6 +1872,28 @@ boolean onNumberKeyInput(int key) {
 	return false;
 }
 
+private StringBuilder validateNewTimeValue(String prefix, String keyValue) {
+	if(typeBuffer.toString() == "") return typeBuffer;
+	int numColons = (prefix.length() - prefix.replace(":", "").length()) / ":".length();
+	String finalVal;
+	//numColons mapping: 0 -> hours, 1 -> minutes, 2-> seconds
+	int val = Integer.valueOf(typeBuffer.toString());
```

Comment:
Please use Integer.parseInt to avoid useless unboxing.

Reply:
Furthermore this tries to parse AM/FM which  leads into yet another NumberFormatException

Reply:
Okay, this seems simple enough to fix, run the validtime check only if SWT.TIME && field != AM_PM. I overlooked this point but it is what it should have been in the first place.